blockchain/state_manager/src/lib.rs

```diff
@@ -138,6 +154,51 @@ where
         &self.cs
     }
 
+    /// Gets 32 bytes of randomness for ChainRand paramaterized by the DomainSeparationTag, ChainEpoch,
+    /// Entropy from the latest beacon entry.
+    pub async fn get_beacon_randomness(
+        &self,
+        blocks: &TipsetKeys,
+        pers: DomainSeparationTag,
+        round: ChainEpoch,
+        entropy: &[u8],
+    ) -> Result<[u8; 32], Box<dyn std::error::Error>> {
+        let chain_rand = ChainRand::new(blocks.to_owned(), self.cs.clone(), self.beacon.clone());
+        match self.get_network_version(round) {
```

Comment:
Would be better to match explicitly all cases of the enum instead of relying on the wildcard pattern (`_`).

Rational: if we introduce a new network version (e.g. `NetworkVersion::V15`), this would fall silently into `get_beacon_randomness_v1` which is probably not what we want. Would be better that the code do not compile and be forced to inspect and handle the new case.

Reply:
good point, updated!